timeseries/src/autogluon/timeseries/learner.py

```diff
@@ -145,7 +131,7 @@ def _align_covariates_with_forecast_index(
 
         If some of the item_ids or timestamps are missing, an exception is raised.
         """
-        if len(self.known_covariates_names) == 0:
+        if len(self.known_covariates_names) == 0 or known_covariates is None:
```

Comment:
This looks potentially buggy to me: If `len(self.known_covariates_names) != 0` but `known_covariates is None` we want to raise an exception (see the next `if` block), but this block will just return `None`.

Reply:
thanks for catching! put up a new version...